fdbclient/NativeAPI.actor.cpp

```diff
@@ -4469,13 +4511,26 @@ Future<Key> Transaction::getKey(const KeySelector& key, Snapshot snapshot) {
 	return getKeyAndConflictRange(cx, key, getReadVersion(), conflictRange, info, options.readTags);
 }
 
-Future<RangeResult> Transaction::getRange(const KeySelector& begin,
-                                          const KeySelector& end,
-                                          GetRangeLimits limits,
-                                          Snapshot snapshot,
-                                          Reverse reverse) {
+template <class GetKeyValuesMaybeHopRequest>
+void increaseCounterForRequest(Database cx) {
+	if constexpr (std::is_same<GetKeyValuesMaybeHopRequest, GetKeyValuesRequest>::value) {
+		++cx->transactionGetRangeRequests;
+	} else if (std::is_same<GetKeyValuesMaybeHopRequest, GetKeyValuesAndHopRequest>::value) {
+		++cx->transactionGetRangeAndHopRequests;
+	} else {
+		UNREACHABLE();
+	}
+}
+
+template <class GetKeyValuesMaybeHopRequest, class GetKeyValuesMaybeHopReply>
```

Comment:
If we want to go with getKeyValueMaybeHop, I'd suggest we don't expose this `getRangeMaybeHop` API (move it to private).

Reply:
Good point. Changed to private.